lib/runtime/transaction_validity.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package runtime
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/transaction"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// TransactionValidityError Information on a transaction's validity and, if valid,
+// on how it relates to other transactions. It is a result of the form:
+// Result<ValidTransaction, TransactionValidityError>
+type TransactionValidityError scale.VaryingDataType
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (tve *TransactionValidityError) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*tve)
+	err = vdt.Set(val)
+	if err != nil {
+		return err
+	}
+	*tve = TransactionValidityError(vdt)
+	return nil
+}
+
+// Value will return the value from the underlying VaryingDataType
+func (tve *TransactionValidityError) Value() (val scale.VaryingDataTypeValue) {
+	vdt := scale.VaryingDataType(*tve)
+	return vdt.Value()
+}
+
+// Error will return the error underlying TransactionValidityError
+func (tve TransactionValidityError) Error() string {
+	if tve.Value() == nil {
+		return "unset TransactionValidityError"
+	}
+	switch err := tve.Value().(type) {
+	case InvalidTransaction:
+		return err.Error()
+	case UnknownTransaction:
+		return err.Error()
+	default:
+		panic(fmt.Sprintf("unexpected value: %T %v", err, err))
+	}
```

Comment:
If we only care to have an `Error()` method, this would be simpler (and also allow to plug in more types in the future without change)

```suggestion
	err, ok := tve.Value().(error)
	if !ok {
		panic(fmt.Sprintf("unexpected value: %T %v", err, err))
	}
	return err.Error()
```